### PR TITLE
Create Ugly_numbers.dart

### PR DESCRIPTION
**Pull request by RimjhimGupta**
This pull request is for Hacktober Fest 2020

# Welcome to Dart community

### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/Ugly_numbers.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  var n;
+  while (a % b == 0) {
+    n = a / b;
+    a = n.toInt();
+  }
+  return a;
+}
+
+/* Function to check if a number is ugly or not */
+int isUgly(int no) {
```
Comment (Parowicz) [inline maths/Ugly_numbers.dart:11]: Such function should return `bool`
```suggestion
bool isUgly(int no) {
```

File: maths/Ugly_numbers.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  no = maxDivide(no, 2);
+  no = maxDivide(no, 3);
+  no = maxDivide(no, 5);
+
+  return (no == 1) ? 1 : 0;
```
Comment (Parowicz) [inline maths/Ugly_numbers.dart:16]:
```suggestion
  return no == 1;
```

File: maths/Ugly_numbers.dart [hunk 3]
```diff
@@ -0,0 +1,37 @@
+int maxDivide(int a, int b) {
```
Comment (Parowicz) [inline maths/Ugly_numbers.dart:1]: You could add some tests
```suggestion
import 'package:test/test.dart';

int maxDivide(int a, int b) {
```

File: maths/Ugly_numbers.dart [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+/* Function to get the nth ugly number*/
+int getNthUglyNo(int n) {
+  int i = 1;
+  int count = 1; /* ugly number count */
+
+/*Check for all integers untill ugly count 
+	becomes n*/
+  while (n > count) {
+    i++;
+    if (isUgly(i) == 1) count++;
```
Comment (Parowicz) [inline maths/Ugly_numbers.dart:28]: After changing `isUgly` return type to `bool` this comparison wont be necessary

File: maths/Ugly_numbers.dart [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  }
+  return i;
+}
+
+/* Driver program to test above functions */
+void main() {
+  var no = getNthUglyNo(150);
+  print('150th ugly no. is $no');
```
Comment (Parowicz) [inline maths/Ugly_numbers.dart:36]: Some tests
```suggestion
  test("getNthUglyNo(150) returns 5832", () {
    expect(getNthUglyNo(150), equals(5832));
  });

  test("isUgly returns true for 6", () {
    expect(isUgly(6), isTrue);
  });

  test("isUgly returns true for 5832", () {
    expect(isUgly(5832), isTrue);
  });

  test("isUgly returns false for 5833", () {
    expect(isUgly(5833), isFalse);
  });
```

**Comment by Parowicz** [comment]
LGTM!

**Comment by RimjhimGupta** [comment]
@Parowicz Sir, I request you to please add label to my pull request for Hacktober Fest 2020.

**Comment by RimjhimGupta** [comment]
@Parowicz Thank you Sir

**Pull request merged**